chia/util/keychain.py

```diff
@@ -315,13 +318,36 @@ def add_private_key(self, mnemonic: str) -> PrivateKey:
             # Prevents duplicate add
             raise KeychainFingerprintExists(fingerprint)
 
-        self.keyring_wrapper.set_passphrase(
-            self.service,
-            get_private_key_user(self.user, index),
-            bytes(key.get_g1()).hex() + entropy.hex(),
-        )
+        # Try to set the label first, it may fail if the label is invalid or already exists.
+        # This can probably just be moved into `FileKeyring.set_passphrase` after the legacy keyring stuff was dropped.
```

Comment:
I agree that we should make this into a single operation of setting the passphrase w/ optional label. If the legacy keyring API is the limiting factor here, I'm ok waiting until we remove that legacy support. But if there's a way to combine the two operations into one now, I think that would be worth pursuing.

Reply:
There is a hacky way maybe like passing the fingerprint down to `FileKeyring` if it doesn't somewhere conflict with the legacy keyring APIs but i would rather not do/try this now since we anyway drop the legacy code soon and it works fine this way for now.